Update PETSc to 3.19 series

cticenhour opened the pull request:
## Reason
MOOSE is currently using the PETSc 3.16 series, while PETSc itself is on the 3.19 series. The PETSc submodule in MOOSE should be updated to the new release. 

## Design
Update PETSc, and fixup any broken MOOSE tests or systems.

## Impact
Shouldn't change core prior functionality of MOOSE, but provides users with new features and bugfixes from the new release.

cticenhour committed:
Modify PETSc configure scripts to satisfy conda build issues and PETSc changes for 3.18

- superlu_dist/cmake missed finding the MPI library for the C compiler unless with-mpi-dir was used
- PETSc 3.18 now prefers determining the C++ dialect itself or via *either* the std flag or dialect configure flag only

cticenhour committed:
Fixup min clang/gcc builds

Superlu_dist requires C++11 be set as dialect for these targets. Setting this dialect explicitly then requires stripping C++17 from CXXFLAGS for PETSc 3.18 when building the conda package (both std and dialect can't be set and passed in at the same time).

cticenhour committed:
Remove ignore from 3.17 changes due to 3.18 bugfixes in PetscDMMoose.C

cticenhour committed:
Reorganize PETSc configure

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 477e44bbb558b1357d86363677accbb4bcdfaabc
+Subproject commit ebf490e5a1712488bc2f86ee394c256c5fd3e19f
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1365571/) on 6136b69 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1365571/) on 6136b69 wanted to post the following:

A change of the following file(s) triggered this check:

  scripts/configure_petsc.sh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1365571/) on 6136b69 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1365571/) on 6136b69 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1365594/) on 6136b69 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23502/site)

This comment will be updated on new commits.

cticenhour committed:
Fixup warnings related to types in LIBMESH_SETERRQ2 in PetscDMMoose.C

- Remove MOOSE PetscInt_FMT shim in favor of libMesh shim
- Cast (un)contacts.size() and displaced.size() to PetscInt

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1375433/) on 5810a11 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1375433/) on 5810a11 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1381727/) on 0b7d72e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1381727/) on 0b7d72e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1384768/) on c5052d4 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1384808/) on c0794db wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1384808/) on c0794db wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1384808/) on c0794db wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1393413/) on 6440312 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1393413/) on 6440312 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1393413/) on 6440312 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Re-gold BlockWeightedPreconditioner replicated mesh test due to changes in PTScotch

This dependency was updated as of PETSc 3.18

cticenhour committed:
modify test spec

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1454706/) on 2ddc64e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1454706/) on 2ddc64e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 477e44bbb558b1357d86363677accbb4bcdfaabc
+Subproject commit c7d19d7b3f2d9e6411c648820d1207320e4154c7
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1471716/) on 99e43ef wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1471716/) on 99e43ef wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1471716/) on 99e43ef wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1480545/) on fac6425 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1480545/) on fac6425 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1480545/) on fac6425 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1600116/) on 801dd86 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1600116/) on 801dd86 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1600116/) on 801dd86 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1660667/) on 2d6ca36 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1660667/) on 2d6ca36 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1660667/) on 2d6ca36 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Fixup distributed mesh failures with PETSc 3.18.6

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1662214/) on 8e08cbb wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>d2ef20</th>
      <th colspan=4>#23502 8e08cb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.59%</td>
      <td>85.59%</td>
      <td>+0.00%</td>
      <td>0.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>89529</td>
      <td>89531</td>
      <td>+2</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15073</td>
      <td>15072</td>
      <td>-1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23502/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23502/coverage/framework)


## Modules coverage

### Optimization

<table>
  <thead>
    <tr>
      <th></th>
      <th>d2ef20</th>
      <th colspan=4>#23502 8e08cb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.58%</td>
      <td>89.14%</td>
      <td>-0.44%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>1316</td>
      <td>1322</td>
      <td>+6</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>153</td>
      <td>161</td>
      <td>+8</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23502/coverage_diff/optimization)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23502/coverage/optimization)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/23502/coverage/xfem)

</details>

## Warnings

- `framework` new line coverage rate 0.00% is less than the suggested 90.0%



This comment will be updated on new commits.

cticenhour committed:
Fixup convergence error check test

cticenhour committed:
Fix typos in versioner script

cticenhour committed:
Update conda to PETSc 3.19 pre-release

Add GNU bison 3.4 as a requirement in moose-mpich (3.18+ now requires bison >= 3.0 for our configuration)

Also adds newsletter entry

Refs #23502

cticenhour committed:
Add cticenhour to mpich, libmesh-vtk, petsc, and libmesh package meta.yaml

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 477e44bbb558b1357d86363677accbb4bcdfaabc
+Subproject commit 7def7855e50b9634803a9910bc790e7aaf084d07
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665514/) on c76a1ff wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665514/) on c76a1ff wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665514/) on c76a1ff wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665560/) on 8316b05 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665560/) on 8316b05 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1665560/) on 8316b05 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented on petsc:
```diff
@@ -1 +1 @@
-Subproject commit 477e44bbb558b1357d86363677accbb4bcdfaabc
+Subproject commit 08c1f94318ee3b3ca1edb31fd14572b8083ef302
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1671480/) on 1f9e622 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1671480/) on 1f9e622 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1671480/) on 1f9e622 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Adjust petsc_version for jacobian inf_nan tests

New PETSc APIs now throw proper divergence errors for NaN values prior to the jacobian evaluation; thus, they are incompatible with this test.

cticenhour committed:
Use PetscAssertPointer in external_petsc_solver

This is deprecated as of 3.20 series

cticenhour committed:
Remove out-of-date PETSc SNESTEST options from shape_side_uo test

cticenhour committed:
Fixup failing porous_flow tests with PETSc 3.19.3 pre-release

New divergence reason status reporting causes jacobian test failures due to dtmin - utilizing abort_on_solve_fail to stop solve on first failure so that jacobian test result can be properly evaluated

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1679522/) on 0a60e55 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1679522/) on 0a60e55 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1679522/) on 0a60e55 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1680456/) on ffbde8e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1680456/) on ffbde8e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1680456/) on ffbde8e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Turn off make check tests for apptainer if openmpi

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1683431/) on fa25905 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1683431/) on fa25905 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1683431/) on fa25905 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1683701/) on b66d86b wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1683701/) on b66d86b wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour reviewed (changes requested):
Need to update https://github.com/idaholab/moose/blob/5f86f198a56ed6985d1ebc282b81da3f049af20b/docker_ci/Dockerfile#L102

cticenhour committed:
Adjust ptscotch_weight_both test tolerance for 32bit dof IDs

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684306/) on 825ecf0 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684306/) on 825ecf0 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684306/) on 825ecf0 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented on modules/doc/content/newsletter/2023/2023_08.md:
````diff
@@ -11,4 +11,31 @@ for a complete description of all MOOSE changes.
+```
+Package                  Version                     Build
+============================================================
+moose-dev                2023.06.23                  build_1
+moose-libmesh            2023.06.23                  build_1
````
```suggestion
moose-dev                2023.07.31                  build_0
moose-libmesh            2023.07.26                  build_1
```

Reminder to do this before merging.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684407/) on d5eafff wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684407/) on d5eafff wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1684407/) on d5eafff wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/1684489/) on d5eafff wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/23502/timing)

This comment will be updated on new commits.

cticenhour committed:
Fixup test/tests/samplers/distribute/plot.py for matplotlib 3.3.2

This is due to zfp needing to be 0.5.5 for strumpack. Conda began resolving matplotlib to 3.3.2, which was triggering errors related to plot.py's use of DataFrames in plot().

cticenhour committed:
Set PETSc option pair options_left to 0 by default

This was turned on by default in PETSc 3.19.3 pre-release (PETSc commit: 59f199a7) for PETSc optimized builds

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1689603/) on eaf672a wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1689603/) on eaf672a wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1689603/) on eaf672a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad committed:
Newer netcdfs require that hdf5 be built with zlib

Refs libmesh/libmesh#3617

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690717/) on 1407a07 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690717/) on 1407a07 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690717/) on 1407a07 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690936/) on 1c60f37 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690936/) on 1c60f37 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1690936/) on 1c60f37 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Add abort_on_solve_fail parameter to TestAction in THM

This resolves failures in RELAP-7 related to #23502

cticenhour commented:
@lindsayad Your commit in #25166 has been moved over as [b58624e](https://github.com/idaholab/moose/pull/23502/commits/b58624ebeb7a3083dd585488f3cebc3e55161ea4). 

Working through app patches currently. RELAP-7 is done pending this PR going in. Griffin, SAM, and Subchannel are on-going.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1698363/) on b58624e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1698363/) on b58624e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1698363/) on b58624e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Fixup tensor mechanics dyn_euler_rayleigh_hht_ti test on Linux parallel testing

cticenhour commented:
I invalidated just in case it was something errant in the environment, but the `zstd` library it noted in the failure is distinct from `zlib` I thought - both compress/decompress algorithms, but different libraries. The change you made to `configure_petsc.sh` is only used when HDF5 needs to be built, which it shouldn't be in the case....we get it from conda here.

lindsayad commented:
So what I'm hearing is that I'm probably not the problem in this case...? 😸

cticenhour commented:
Doesn't seem so :) the invalidate did the trick in this case!

YaqiWang commented:
Any help for fixing Griffin? All failing tests seem close related.

cticenhour commented:
@YaqiWang if you'd like to help out with Griffin I would welcome it! I haven't been able to get to it quite yet.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700401/) on 064353f wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700401/) on 064353f wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Strumpack is not completely valgrind clean, so skip hmg_strumpack test for now

See https://github.com/pghysels/STRUMPACK/issues/103 for more context

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700850/) on ef0828e wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700850/) on ef0828e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700850/) on ef0828e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700984/) on 7da642c wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1700984/) on 7da642c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Fixup solution_invalid tests failing in parallel with petsc 3.19.3

Refs #23502

cticenhour committed:
Add context to fixups in porous_flow tests

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1705664/) on ce0757c wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1705664/) on ce0757c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour committed:
Change preconditioners.hmg tests to use GAMG with HMG

This works around breaking changes to hypre for this case introduced in PETSc commit 06977982913c3aebee464d9f4c9f5c3699a59e57 within PETSc PR 6651

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720276/) on 43c43eb wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720276/) on 43c43eb wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1720276/) on 43c43eb wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@grmnptr and @zachmprince -- Pinging you both for help if you have some time. I was working this weekend on finishing off the last failing MOOSE test in stochastic tools heavy, and discovered *what* has changed in the PETSc TaO interface to cause the test failure, but I'm at a loss as to what the proper fix is (mostly due to unfamiliarity with TaO and the ST module itself). To give some background, the failing test is `examples/surrogates/gaussian_process.4D_HC_avg/train`, which seems (initially) to train well, but fails during the test of the calculated length factors. 

I initially bisected this back to [this PETSc commit in 3.18.0](https://gitlab.com/petsc/petsc/-/commit/9203fd1f6b645bbbc6de9e81e30dd3078b115bfc), which ensures consistency in the TaO linesearch `stepmax` and `stepmin` for every call to `TaoLineSearchApply_MT` by backing them up prior to the calculation of a bounded linesearch step and resetting them before finishing the computations. This wasn't sufficient to explain the differing behavior in the failing test, so I bisected again to find [this commit in 3.19.0](https://gitlab.com/petsc/petsc/-/commit/def90ec86b14b00b9b40b25f336eecd3d40eae8c), adding a failure point in this same section of code, which was already failing to converge (I will attach some TaO output in a dropdown section of my post below). Reverting both of these changes does indeed allow the test pair to pass successfully. 

My attempts to fix this involves changes to `stepmax` and `stepmin` to attempt to keep them away from a zero step, as well as to keep them within a prior hard-coded maximum dictated by the original commit I highlighted above. Tolerance adjustments also did not seem to have a demonstrable impact. Attempting other combinations of linesearch objects and CG types was also unsuccessful. Also one can see in the test CSV output that the training simulation consistently hits the rails of the `tuning_min` and `tuning_max` parameters in at least some of the calculated length factors. 

Take a look at the dropdown below to see some relevant output from the training test that is not performing correctly. I would really appreciate any help that either of you can provide!

<details>
<summary>TaO output using `show_optimization_details = true` and `tao_monitor`:</summary>
<br>
Details GP_normal_mc.i output for PR PETSc submodule:

```
   0 TAO,  Function value: 709.469,  Residual: 0.253798
   0 TAO,  Function value: 709.469,  Residual: 0.139531
   1 TAO,  Function value: 634.068,  Residual: 234.16
 Tao Object: 1 MPI process
   type: bncg
     CG Type: pcgd
     Skipped Stepdirection Updates: 0
     Scaled gradient steps: 1
     Pure gradient steps: 2
     Not a descent direction: 0
     Line search fails: 3
   TaoLineSearch Object: 1 MPI process
     type: more-thuente
     maximum function evaluations=30
     tolerances: ftol=0.0001, rtol=1e-10, gtol=0.9
     total number of function evaluations=0
     total number of gradient evaluations=0
     total number of function/gradient evaluations=1
     using variable bounds
     Termination reason: 6
   Active Set subset type: subvec
   convergence tolerances: gatol=1e-08,   steptol=0.,   gttol=0.
   Residual in Function/Gradient:=234.16
   Objective value=634.068
   total number of iterations=2,                          (max: 2000)
   total number of function/gradient evaluations=12,      (max: 4000)
   Solver terminated: -6   Line Search Failure
 Size	global =  5		local =  5
 #	Value
 0	1.18696
 1	0.680053
 2	0.206921
 3	0.749758
 4	0.379848
 ```

Details GP_normal_mc.i output with reverted changes:

```
   0 TAO,  Function value: 709.469,  Residual: 0.253798
   0 TAO,  Function value: 709.469,  Residual: 0.139531
   1 TAO,  Function value: 634.068,  Residual: 234.16
   2 TAO,  Function value: 427.762,  Residual: 292.089
 Tao Object: 1 MPI process
   type: bncg
     CG Type: pcgd
     Skipped Stepdirection Updates: 0
     Scaled gradient steps: 1
     Pure gradient steps: 2
     Not a descent direction: 0
     Line search fails: 3
   TaoLineSearch Object: 1 MPI process
     type: more-thuente
     maximum function evaluations=30
     tolerances: ftol=0.0001, rtol=1e-10, gtol=0.9
     total number of function evaluations=0
     total number of gradient evaluations=0
     total number of function/gradient evaluations=2
     using variable bounds
     Termination reason: 3
   Active Set subset type: subvec
   convergence tolerances: gatol=1e-08,   steptol=0.,   gttol=0.
   Residual in Function/Gradient:=292.089
   Objective value=427.762
   total number of iterations=3,                          (max: 2000)
   total number of function/gradient evaluations=19,      (max: 4000)
   Solver terminated: -6   Line Search Failure
 Size	global =  5		local =  5
 #	Value
 0	1.18032
 1	0.232354
 2	0.654185
 3	0.897953
 4	0.696113
 ```
 
</details>

grmnptr commented:
I can take a look at this tomorrow!

grmnptr commented:
Looks like the iteration values are the same, the new version just terminates one iteration early. Hmmm, I wonder if this is tolerance related.

grmnptr commented:
Yeah, convergence result is:
(3) for the old one which is hitting the `gatol`
(6) for the now one which is hotting the `steptol`
Not sure how to change these, but we will see tomorrow

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1737624/) on 310ffc7 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1737624/) on 310ffc7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1738567/) on 70c9388 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1738567/) on 70c9388 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1738567/) on 70c9388 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@YaqiWang were you able to make progress with a Griffin patch? With the closing of #25360, we only have app patches to go here.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1744366/) on 6468e25 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1744366/) on 6468e25 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1744366/) on 6468e25 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753267/) on 3c795e3 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753267/) on 3c795e3 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1753267/) on 3c795e3 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Controlled app tests](https://civet.inl.gov/job/1753291/) on 3c795e3 : invalidated by @lindsayad

cticenhour commented:
@lindsayad will need to rebase this tomorrow in order to get an up-to-date CI result.

cticenhour committed:
Update versioner hashes (#23502)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1762253/) on 7738b87 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1762253/) on 7738b87 wanted to post the following:

A change of the following file(s) triggered this check:

  scripts/configure_petsc.sh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1762253/) on 7738b87 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1762253/) on 7738b87 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

cticenhour commented:
@lindsayad Subchannel patch is [ncrc/subchannel#294](https://github.inl.gov/ncrc/subchannel/pull/294), and is ready for a submodule update when we are ready to merge this. Is there a SAM patch ready?

lindsayad commented:
No I will have to look at SAM

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1770926/) on e529182 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1770926/) on e529182 wanted to post the following:

A change of the following file(s) triggered this check:

  scripts/configure_petsc.sh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1770926/) on e529182 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1770926/) on e529182 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented on scripts/configure_petsc.sh:
```diff
@@ -69,9 +69,11 @@ function configure_petsc()
   if [ -z "$HDF5_STR" ]; then
     HDF5_STR='--download-hdf5=1'
     HDF5_FORTRAN_STR='--download-hdf5-fortran-bindings=0'
+    HDF5_CONFIGURE_STR='--with-zlib'
```
```suggestion
    HDF5_CONFIGURE_STR='--download-hdf5-configure-arguments="--with-zlib"'
```
This must have been a mistake in my commit

cticenhour replied:
Will fixup the original commit.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775051/) on e76fc68 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775051/) on e76fc68 wanted to post the following:

A change of the following file(s) triggered this check:

  scripts/configure_petsc.sh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775051/) on e76fc68 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775051/) on e76fc68 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775194/) on 86ed489 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775194/) on 86ed489 wanted to post the following:

A change of the following file(s) triggered this check:

  scripts/configure_petsc.sh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The PETSc submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1775194/) on 86ed489 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Controlled app tests](https://civet.inl.gov/job/1775217/) on 86ed489 : invalidated by @cticenhour

Subchannel patched

cticenhour commented:
Thanks @lindsayad! Apptainer build cuda is a known failure, and has temporarily been disabled in idaholab/civet_recipes#1457. Other failures are unrelated or allowed to fail pending independent fixup from application leads. 

@loganharbour - re-requesting a review from you, pending the completion of the CI package builds.

loganharbour commented:
Running them now. We don't need them to be complete, I just want a sanity check on if one build works :D

loganharbour commented on conda/petsc/build.sh:
```diff
@@ -36,18 +36,19 @@ else
-    FC="$FC" \
-    F90="$F90" \
-    F77="$F77" \
+    --with-mpi-dir=$PREFIX \
```
How does this work?

cticenhour replied:
MPI (via `moose-mpich`) is installed in the `$PREFIX` location as defined by `conda-build`. This is the temporary environment that build creates and then PETSC is also installed to (see `--prefix=$PREFIX` a few lines down in the same file). Then packaging occurs and generalizes this. This only impacts the build and installation - conda takes care of the rest. 

Does that make sense?

loganharbour replied:
oh, absolutely. I would have gotten that had I looked at the path... whoops

cticenhour replied:
Using $PREFIX here still maintains consistency with the end-user environment. This PETSc will always use the MPI provided by conda.

loganharbour reviewed (approved):
Good to go. One of them built, which is good enough for me. CUDA container is related, but ignored for now. We've decided to stop building this container as its use is null at the moment, it needs more work, and this is more important.
